From a0c25b8f4627f0331108b3452273859b71bbc9f6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Torleif=20Sk=C3=A5r?= <torleif.skaar@gmail.com>
Date: Thu, 24 Jul 2025 14:58:01 +0200
Subject: [PATCH] makefile: Ensure that 'OBJS_DIR' exist for all dependencies

It looks like there is some racy-ness here causing
issues with the copying of files to the objs dir,
causing the scmos techfiles to not get copied
---
 scmos/cif_template/Makefile | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/scmos/cif_template/Makefile b/scmos/cif_template/Makefile
index 89377a82..c068f27c 100644
--- a/scmos/cif_template/Makefile
+++ b/scmos/cif_template/Makefile
@@ -32,35 +32,35 @@ $(OBJS_DIR):
 	$(MKDIR) $(OBJS_DIR)
 
 $(CIFIN): cifin.c cifin-cmos26b.gen cifin.gen cifin.nw cifin.oldnw\
-	  cifin.others cifin.pw cifin-ami16.gen
+	  cifin.others cifin.pw cifin-ami16.gen | $(OBJS_DIR)
 	rm -f $(CIFIN)
 	$(SC_CPP) -DSTANDARD cifin.c | ${SED_CMD} > $(CIFIN)
 
 $(CIFOUT): cifout.c cifout-cmos26b.gen cifout.gen cifout.nw\
-	   cifout.others cifout.pw cifout-ami16.gen
+	   cifout.others cifout.pw cifout-ami16.gen | $(OBJS_DIR)
 	rm -f $(CIFOUT)
 	$(SC_CPP) -DSTANDARD cifout.c | ${SED_CMD} > $(CIFOUT)
 
-$(IBMCIFIN): cifin.c cifin-ibm.gen
+$(IBMCIFIN): cifin.c cifin-ibm.gen | $(OBJS_DIR)
 	rm -f $(IBMCIFIN)
 	$(SC_CPP) -DIBM cifin.c | ${SED_CMD} > $(IBMCIFIN)
 
-$(IBMCIFOUT): cifout.c cifout-ibm.gen
+$(IBMCIFOUT): cifout.c cifout-ibm.gen | $(OBJS_DIR)
 	rm -f $(IBMCIFOUT)
 	$(SC_CPP) -DIBM cifout.c | ${SED_CMD} > $(IBMCIFOUT)
 
-$(TMCIFIN): cifin.c cifin-cmos26b.gen
+$(TMCIFIN): cifin.c cifin-cmos26b.gen | $(OBJS_DIR)
 	rm -f $(TMCIFIN)
 	$(SC_CPP) -DTIGHTMETAL cifin.c | ${SED_CMD} > $(TMCIFIN)
 
-$(TMCIFOUT): cifout.c cifout-cmos26b.gen
+$(TMCIFOUT): cifout.c cifout-cmos26b.gen | $(OBJS_DIR)
 	rm -f $(TMCIFOUT)
 	$(SC_CPP) -DTIGHTMETAL cifout.c | ${SED_CMD} > $(TMCIFOUT)
 
-$(SUBCIFIN): cifin.c cifin-cmos26g.gen cifin-cmos14b.gen
+$(SUBCIFIN): cifin.c cifin-cmos26g.gen cifin-cmos14b.gen | $(OBJS_DIR)
 	rm -f $(SUBCIFIN)
 	$(SC_CPP) -DSUBMICRON cifin.c | ${SED_CMD} > $(SUBCIFIN)
 
-$(SUBCIFOUT): cifout.c cifout-cmos26g.gen cifout-cmos14b.gen
+$(SUBCIFOUT): cifout.c cifout-cmos26g.gen cifout-cmos14b.gen | $(OBJS_DIR)
 	rm -f $(SUBCIFOUT)
 	$(SC_CPP) -DSUBMICRON cifout.c | ${SED_CMD} > $(SUBCIFOUT)
-- 
2.50.1

